### PR TITLE
Reverse 2878

### PR DESCRIPTION
**Pull request by djkirkham**
A workaround for https://github.com/dask/dask/issues/2823 was added in https://github.com/SciTools/iris/pull/2878. The dask issue is fixed in the latest release, so this PR reverts the changes.

N.B. There is a test which will fail on master now that the Dask issue is fixed, so this needs resolving ASAP.

### CODE REVIEW

File: lib/iris/_lazy_data.py [hunk 1]
```diff
@@ -104,15 +95,10 @@ def as_lazy_data(data, chunks=None, asarray=False):
         # but reduce it if larger than a default maximum size.
         chunks = _limited_shape(data.shape)
 
+    if isinstance(data, ma.core.MaskedConstant):
+        data = ma.masked_array(data.data, mask=data.mask)
```
Comment (pp-mo) [inline lib/iris/_lazy_data.py]: Is this related, or fixing a different problem ?

File: lib/iris/_lazy_data.py [hunk 2]
```diff
@@ -138,7 +124,10 @@ def as_concrete_data(data):
         # In some cases dask may return a scalar numpy.int/numpy.float object
         # rather than a numpy.ndarray object.
         # Recorded in https://github.com/dask/dask/issues/2111.
+        dtype = data.dtype
         data = np.asanyarray(data.compute())
+        if isinstance(data, ma.core.MaskedConstant):
```
Comment (pp-mo) [inline lib/iris/_lazy_data.py]: Again, how is this tied up with the main point ?

File: lib/iris/_lazy_data.py [hunk 3]
```diff
@@ -138,7 +124,10 @@ def as_concrete_data(data):
         # In some cases dask may return a scalar numpy.int/numpy.float object
         # rather than a numpy.ndarray object.
         # Recorded in https://github.com/dask/dask/issues/2111.
+        dtype = data.dtype
```
Comment (pp-mo) [inline lib/iris/_lazy_data.py]: The implication here is that `array.dtype == array.compute().dtype` is not guaranteed.
Is that really true + if so why ??

File: lib/iris/_lazy_data.py (hunk 1 above)
Comment (djkirkham) [reply lib/iris/_lazy_data.py]: This was the way the code was before the original change. I had moved it into the `_getall` function, which I've now removed. I _think_ it's necessary because dask doesn't like `MaskedConstant`.

File: lib/iris/_lazy_data.py (hunk 3 above)
Comment (djkirkham) [reply lib/iris/_lazy_data.py]: If the result of the compute is a `MaskedConstant`, it will always have a float dtype

File: lib/iris/_lazy_data.py (hunk 2 above)
Comment (djkirkham) [reply lib/iris/_lazy_data.py]: When I reverted the code to how it was before I found that some of our tests which check that the result of realising lazy data is not a `MaskedConstant` were failing. It seems that Dask is returning a `MaskedConstant` in situations where it wasn't before.

**Comment by pp-mo** [comment]
> the way the code was before the original change

Ah, thanks.  Stupidly, I hadn't thought to compare with the pre-change code.
That's all good, then.
